sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.models.ts

```diff
@@ -0,0 +1,310 @@
+  })
+  applicationStatus: ApplicationStatus;
+  /**
+   * Application archived.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsBoolean({
+    message: `${userFriendlyNames.isArchived} must be a boolean value.`,
+  })
+  @IsIn([false], {
+    message: `Application is already ${userFriendlyNames.archived} and cannot be withdrawn.`,
+  })
+  isArchived: boolean;
+  /**
+   * Has previously been withdrawn.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsBoolean({
+    message: `The ${userFriendlyNames.hasPreviouslyBeenWithdrawn} must be a boolean value.`,
+  })
+  @IsIn([false], {
+    message: (validationArguments: ValidationArguments) =>
+      `The ${userFriendlyNames.application} is already withdrawn with the date: ${validationArguments.object["withdrawalDate"]}.`,
+    context: ValidationContext.CreateWarning(
+      ApplicationWithdrawalValidationWarnings.HasPreviouslyBeenWithdrawn,
+    ),
+  })
+  hasPreviouslyBeenWithdrawn: boolean;
+  /**
+   * Is record match .ie. SIN, application number and institution must match a record in database.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsBoolean({
+    message: `${userFriendlyNames.recordMatch} must be a boolean value.`,
+  })
+  @IsIn([true], {
+    message: `${userFriendlyNames.record} and will be skipped.`,
+    context: ValidationContext.CreateWarning(
+      ApplicationWithdrawalValidationWarnings.NoMatchingRecordFound,
+    ),
+  })
+  isRecordMatch: boolean;
+}
+
+/**
+ * Application bulk withdrawal header and data models.
+ */
+export interface FileData {
+  header: ApplicationBulkWithdrawalHeader;
+  applicationWithdrawalModels: ApplicationWithdrawalImportTextModel[];
+}
+
+/**
+ * Validation warning with a unique type and
+ * a user-friendly message to be displayed.
+ */
+export interface ValidationResult {
+  typeCode: ApplicationWithdrawalValidationWarnings;
+  message: string;
+}
+
+/**
+ * Results of the application bulk withdrawal model validation.
+ */
+export interface ApplicationWithdrawalValidationResult {
+  /**
+   * Record index in the list of records. Headers are not considered.
+   */
+  index: number;
+  /**
+   * validated application bulk withdrawal model.
+   */
+  applicationBulkWithdrawalBusinessValidationModel: ApplicationBulkWithdrawalImportBusinessValidationModel;
+  /**
+   * Warnings, if any.
+   */
+  warnings: ValidationResult[];
+  /**
+   * Users friendly errors list, if any.
+   */
+  errors: string[];
+}
+
+export interface ApplicationData {
+  sin: string;
+  locationId: number;
+  locationCode: string;
+  applicationStatus: ApplicationStatus;
+  isArchived: boolean;
+  hasPreviouslyBeenWithdrawn: boolean;
+}
```

Comment:
None of those properties would be potentially `null`?

Reply:
Nope, I don't think so.